cms/djangoapps/contentstore/views/course.py

```diff
@@ -869,86 +869,65 @@ def _remove_tab(request, tab_type, course_module):
-    is a hack that does the following :
-    1) adds/removes the open ended panel tab to a course automatically
-    if the user has indicated that they want to edit the
-    combinedopendended or peergrading module
-    2) adds/removes the notes panel tab to a course automatically if
-    the user has indicated that they want the notes module enabled in
-    their course
-    """
-    # TODO refactor the above into distinct advanced policy settings
-    filter_tabs = True  # Exceptional conditions will pull this to False
-    if ADVANCED_COMPONENT_POLICY_KEY in request.json:  # Maps tab types to components
-        tab_component_map = {
-            'open_ended': OPEN_ENDED_COMPONENT_TYPES,
-            'notes': NOTE_COMPONENT_TYPES,
-        }
-        # Check to see if the user instantiated any notes or open ended components
-        for tab_type in tab_component_map.keys():
-            component_types = tab_component_map.get(tab_type)
-            found_ac_type = False
-            for ac_type in component_types:
-                # Check if the user has incorrectly failed to put the value in an iterable.
-                new_advanced_component_list = request.json[ADVANCED_COMPONENT_POLICY_KEY]['value']
-                if hasattr(new_advanced_component_list, '__iter__'):
-                    if ac_type in new_advanced_component_list and ac_type in ADVANCED_COMPONENT_TYPES:
-                        if _add_tab(request, tab_type, course_module):
-                            # Set this flag to avoid the tab removal code below.
-                            filter_tabs = False
-                        found_ac_type = True  # break
-                else:
-                    # If not iterable, return immediately and let validation handle.
-                    return
+def _a_component_is_present(request, advanced_components):
+    """
+    Return True when one of `advanced_components` is to be set by request.
 
-            # If we did not find a module type in the advanced settings,
-            # we may need to remove the tab from the course.
-            if not found_ac_type:  # Remove tab from the course if needed
-                if _remove_tab(request, tab_type, course_module):
-                    # Indicate that tabs should *not* be filtered out of
-                    # the metadata
-                    filter_tabs = False
+    raises TypeError
+    when request.ADVANCED_COMPONENT_POLICY_KEY is malformed (not a list)
+    """
+    if ADVANCED_COMPONENT_POLICY_KEY not in request.json:
+        return False
 
-    return filter_tabs
+    new_advanced_component_list = request.json[ADVANCED_COMPONENT_POLICY_KEY]['value']
+    for ac_type in advanced_components:
+        if ac_type in new_advanced_component_list and ac_type in ADVANCED_COMPONENT_TYPES:
+            return True
 
 
 # pylint: disable=invalid-name
-def _config_course_settings(request, course_module, filter_tabs=True):
+def _component_value_true(request, component_types):
```

Comment:
Do we really need to pass the request as an argument or we can pass `request.json` instead?